test/integration/cpp/microbench.cpp

```diff
@@ -0,0 +1,68 @@
+#include "test/jemalloc_test.h"
+#include "test/bench.h"
+
+static void
+new_free(void) {
+	auto p = ::operator new(4);
+	expect_ptr_not_null(p, "Unexpected new failure");
+	free(p);
+}
+
+static void
+new_delete(void) {
+	/* 
+ 	 * Using the trick below to new a void* so that delete, 
+ 	 * instead of sized delete, will be called.
+ 	 */
+	auto p = ::operator new(4);
+	expect_ptr_not_null(p, "Unexpected new failure");
+	::operator delete(p);
+}
+
+static void
+new_free_array(void) {
+	auto p = new int(5);
+	expect_ptr_not_null(p, "Unexpected new[] failure");
+	free(p);
+}
+
+static void
+new_delete_array(void) {
+	auto p = new int(5);
+	expect_ptr_not_null(p, "Unexpected new[] failure");
+	delete[] p;
+}
+
+static void
+new_sized_delete(void) {
+	auto p = ::operator new(4);
+	expect_ptr_not_null(p, "Unexpected new failure");
+	/* Call sized delete explicitly. */
+	::operator delete(p, 4);
+}
+
+TEST_BEGIN(test_free_vs_delete) {
+	compare_funcs(10*1000*1000, 100*1000*1000, "free", (void *)new_free,
+	    "delete", (void *)new_delete);
+}
+TEST_END
+
+TEST_BEGIN(test_free_array_vs_delete_array) {
+	compare_funcs(10*1000*1000, 100*1000*1000, "free_array", 
+	    (void *)new_free_array, "delete_array", (void *)new_delete_array);
+}
+TEST_END
+
+TEST_BEGIN(test_delete_vs_sized_delete) {
+	compare_funcs(10*1000*1000, 100*1000*1000, "delete", 
+	    (void *)new_delete, "sized_delete", (void *)new_sized_delete);
+}
+TEST_END
+
+int
+main() {
+	return test(
```

Comment:
For benchmarks, no need to test different reentrancy cases (i.e right now each test is done 3 times). You can use ` test_no_reentrancy()` here.

Reply:
Modified